### PR TITLE
Report ZuulExceptions through logger in ZuulServletFilter

### PR DESCRIPTION
**Pull request by ghenkes**
In ZuulServletFilter ZuulExceptions are just getting printed rather than getting added to the log like in ZuulServlet.

### CODE REVIEW

File: zuul-core/src/main/java/com/netflix/zuul/filters/ZuulServletFilter.java [hunk 1]
```diff
@@ -109,7 +113,7 @@ void init(HttpServletRequest servletRequest, HttpServletResponse servletResponse
     void error(ZuulException e) {
         RequestContext.getCurrentContext().setThrowable(e);
         zuulRunner.error();
-        e.printStackTrace();
+        LOG.error(e.getMessage(), e);
```
Comment (spencergibb) [inline zuul-core/src/main/java/com/netflix/zuul/filters/ZuulServletFilter.java]: My preference would be to remove `e.printStackTrace()` altogether and let the 'error' filters determine if they want to log the error.

Comment (NiteshKant) [inline zuul-core/src/main/java/com/netflix/zuul/filters/ZuulServletFilter.java]: +1

**Comment by ghenkes** [comment]
@spencergibb @NiteshKant removed logging the exception in ZuulServletFilter. Also removed similar line in ZuulServlet. Added logging to the example ErrorResponse filter in zuul-netfilx-webapp.

**Comment by NiteshKant** [comment]
Thanks @ghenkes !

**Pull request merged**